src/window-service.js

```diff
@@ -94,26 +94,34 @@
     }
 
     class DragService {
-        constructor(storeService, geometryService, windowTracker, tearoutWindow) {
+        constructor(storeService, geometryService, windowTracker, tearoutWindow, $q) {
             this.storeService = storeService;
             this.geometryService = geometryService;
             this.windowTracker = windowTracker;
             this.tearoutWindow = tearoutWindow;
+            this.$q = $q;
             this.otherInstance = null;
         }
 
-        overAnotherInstance() {
-            var mainWindows = this.windowTracker.getMainWindows();
+        overAnotherInstance(cb) {
+            var mainWindows = this.windowTracker.getMainWindows(),
+                result = false,
+                promises = [];
 
-            for (var i = 0, max = mainWindows.length; i < max; i++) {
-                var mainWindow = mainWindows[i];
-                if (this.geometryService.windowsIntersect(this.tearoutWindow, mainWindow.getNativeWindow())) {
-                    this.otherInstance = mainWindow;
-                    return true;
-                }
-            }
+            mainWindows.forEach(mainWindow => {
+                var deferred = this.$q.defer();
+                promises.push(deferred.promise);
+                mainWindow.getState(state => {
+                    if (state !== 'minimized' && this.geometryService.windowsIntersect(this.tearoutWindow, mainWindow.getNativeWindow())) {
+                        this.otherInstance = mainWindow;
+                        result = true;
+                    }
 
-            return false;
+                    deferred.resolve();
+                });
+            });
+
+            this.$q.all(promises).then(() => cb(result));
```

Comment:
The native `Promise.race` might be a good thing to try, I don't think it'll make much difference aside from running time.